### PR TITLE
Initial commit of project setup script

### PR DESCRIPTION
**Pull request by froi**
*Note*: Before submitting this pull request, please review our [contributing guidelines](https://github.com/TheIndexingProject/contratospr-api/blob/master/CONTRIBUTING.md).

## Description

This is the initial commit for the project setup script. This script intends to give new users and/or contributors an easier time setting up this project.

The script automates the steps found in out [project setup section of our CONTRIBUTING.md](https://github.com/TheIndexingProject/contratospr-api/blob/master/CONTRIBUTING.md#project-setup)

### Running the script

To run this script you just have to execute it from the root of the project `./project-setup`

### CODE REVIEW

**Comment by froi** [comment]
I'm getting an error when doing the project setup when applying the migrations.

```
Traceback (most recent call last):
  File "manage.py", line 18, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 234, in handle
    fake_initial=fake_initial,
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 246, in apply_migration
    self.record_migration(migration)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 261, in record_migration
    self.recorder.record_applied(migration.app_label, migration.name)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/recorder.py", line 83, in record_applied
    self.migration_qs.create(app=app, name=name)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 422, in create
    obj.save(force_insert=True, using=self.db)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/base.py", line 741, in save
    force_update=force_update, update_fields=update_fields)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/base.py", line 779, in save_base
    force_update, using, update_fields,
  File "/usr/local/lib/python3.7/site-packages/django/db/models/base.py", line 870, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/base.py", line 908, in _do_insert
    using=using, raw=raw)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 1186, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1335, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 99, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.IntegrityError: null value in column "id" violates not-null constraint
DETAIL:  Failing row contains (null, contracts, 0005_auto_20190825_1958, 2019-10-26 20:48:59.360842+00).
```

I'm thinking that our resent changes might have made our database dump (backup.dump) obsolete.

I'll post more as I test this.

**Comment by froi** [comment]
The issues seem to be with the `backup.dump`. When using this file to restore the database it is not including the sequence needed in the internal django migrations table.

I'm looking into how to solve this correctly.

**Comment by froi** [comment]
Adding the `blocked` label until we resolve #91

**Comment by froi** [comment]
Got the setup script to work correctly.

__T.L.D.R.__ 

Docker volumes created by docker-compose prevented us from using the `backup.dump` file. Full explanation is here https://github.com/TheIndexingProject/contratospr-api/issues/91#issuecomment-546796799

Here's a gif of the setup script running, cause why not:

setup-script-run

@jpadilla Can I bother you for a review before I merge this puppy 🐶

File: project-setup [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    echo ""
+    read -p "Do you want to run this in detached mode (will not show logs)? [Y/n]" choice
+    echo ""
+
+    echo "Building Docker images. 🐳🐳🐳"
+
+    if [ $choice == "Y" ] || [ $choice == "y" ] ; then
+        docker-compose up --build -d
+    else
+        docker-compose up --build
```
Comment (jpadilla) [inline project-setup]: If running in the foreground, will the rest of this script execute?

File: project-setup [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+pipenv run pre-commit install
+
+echo ""
+
+echo "Creating the .env file. Remember to change this file with your settings."
+if [ -f example.env ]; then
+    cp example.env .env
+else
+    echo "PYTHONUNBUFFERED=1" >> .env
```
Comment (jpadilla) [inline project-setup]: I’d remove this else branch. We’re better off getting rid of the .env for development and move these inside the docker-compose. When we do, will be able to remove this whole part too.

File: project-setup [hunk 3]
```diff
@@ -0,0 +1,181 @@
+#! /bin/bash
+
+echo "Thanks for your interest in ContratosPR and its API ❤️"
+echo "This script is meant to help you with the setup process for this project that is outlined " \
+    "here https://github.com/TheIndexingProject/contratospr-api/blob/master/CONTRIBUTING.md#project-setup. 👀"
```
Comment (jpadilla) [inline project-setup]: Could we move all URLs to variables in the top of this file?

File: project-setup (hunk 1 above)
Comment (froi) [reply project-setup]: Excellent point. I think it should just be removed.

File: project-setup (hunk 2 above)
Comment (froi) [reply project-setup]: Fair point.

**Pull request merged**